### PR TITLE
Feature/databrick configuration

### PR DESCRIPTION
**Pull request by satenderrathee**

### CODE REVIEW

File: deploy/azure/infra/main.tf [hunk 1]
```diff
@@ -241,3 +241,30 @@ resource "azurerm_role_assignment" "adb_role" {
...
+}
+
+resource "azurerm_key_vault_secret" "databricks-host" {
+  name         = "databricks-host"
+  value        = module.adb.databricks_hosturl
+  key_vault_id = module.kv_default.id
+}
+
+resource "databricks_secret_scope" "kv" {
+  name = "keyvault-managed"
```
Comment (motylp) [inline deploy/azure/infra/main.tf:264]:
```suggestion
  name = "key-vault-managed"
```

Comment (motylp) [reply deploy/azure/infra/main.tf]: thanks 😃

**Review by motylp — APPROVED** [review]

File: deploy/azure/infra/main.tf [hunk 2]
```diff
@@ -241,3 +241,30 @@ resource "azurerm_role_assignment" "adb_role" {
...
 }
+
+resource "databricks_token" "pat" {
+  comment  = "Terraform Provisioning"
+  // 120 day token
+  lifetime_seconds = 120 * 24 * 60 * 60
+}
+
+resource "azurerm_key_vault_secret" "databricks_token" {
+  name         = "databricks-token"
```
Comment (Trishisingh) [inline deploy/azure/infra/main.tf:252]: can this be a variable

File: deploy/azure/infra/main.tf [hunk 3]
```diff
@@ -241,3 +241,30 @@ resource "azurerm_role_assignment" "adb_role" {
...
+  value        = databricks_token.pat.token_value
+  key_vault_id = module.kv_default.id
+}
+
+resource "azurerm_key_vault_secret" "databricks-host" {
+  name         = "databricks-host"
```
Comment (Trishisingh) [inline deploy/azure/infra/main.tf:258]: can this be a variable?

File: deploy/azure/infra/main.tf [hunk 4]
```diff
@@ -241,3 +241,30 @@ resource "azurerm_role_assignment" "adb_role" {
...
+  value        = module.adb.databricks_hosturl
+  key_vault_id = module.kv_default.id
+}
+
+resource "databricks_secret_scope" "kv" {
+  name = "key-vault-backed"
```
Comment (Trishisingh) [inline deploy/azure/infra/main.tf:264]: variable?

File: deploy/azure/infra/main.tf (hunk 2 above)
Comment (satenderrathee) [reply deploy/azure/infra/main.tf]: updated

File: deploy/azure/infra/main.tf (hunk 3 above)
Comment (satenderrathee) [reply deploy/azure/infra/main.tf]: updated

File: deploy/azure/infra/main.tf (hunk 4 above)
Comment (satenderrathee) [reply deploy/azure/infra/main.tf]: done

File: deploy/azure/infra/vars.tf [hunk 5]
```diff
@@ -324,6 +324,29 @@ variable "databricks_group_display_name" {
...
+variable "databricks_secret_scope_kv" {
+  type        = string
+  default     = "key-vault-backed"
+  description = "Name of the databricks secret scope for Key vault."
+}
+
+variable "databricks_pat_comment" {
+  type        = string
+  default     = "Terraform Provisioning"
+  description = "Comment for databricks PAT"
```
Comment (motylp) [inline deploy/azure/infra/vars.tf:348]: What's PAT? is it an environment?

Comment (satenderrathee) [reply deploy/azure/infra/vars.tf:348]: its databricks Personal access tokens

**Review by Trishisingh — APPROVED** [review]

**Pull request merged**